### PR TITLE
BUG: Series[bool].__setitem__(scalar, non_bool)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -1910,6 +1910,9 @@ def validate_numeric_casting(dtype: np.dtype, value: Scalar) -> None:
             raise ValueError(
                 f"Cannot assign {type(value).__name__} to float/integer series"
             )
+    if dtype.kind == "b":
```
Comment (jreback) [inline pandas/core/dtypes/cast.py:1913]: elif

File: pandas/core/dtypes/cast.py [hunk 2]
```diff
@@ -1910,6 +1910,9 @@ def validate_numeric_casting(dtype: np.dtype, value: Scalar) -> None:
...
+        if is_scalar(value) and not is_bool(value):
```
Comment (jreback) [inline pandas/core/dtypes/cast.py:1914]: worth making these not compound? both L1913-1914 and L1907-1908 ?

Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py:1914]: i dont think so; i expect were going to have to flesh out the dtype.kind == "b" case for non-scalars before too long (hopefully can dedup with can_hold_element)

File: pandas/core/dtypes/cast.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py:1913]: sure

**Pull request merged**